server/middleware/checkPopulateUserCaseloads.ts

```diff
@@ -0,0 +1,63 @@
+import { RequestHandler, Router } from 'express'
+import { validate as uuidValidate } from 'uuid'
+import logger from '../../logger'
+import PrisonApiService from '../services/prisonApi/prisonApiService'
+import CsipApiService from '../services/csipApi/csipApiService'
+import { ServiceConfigInfo } from '../services/csipApi/csipApiClient'
+import { CaseLoad } from '../interfaces/caseLoad'
+
+export default function checkPopulateUserCaseloads(
+  prisonApiService: PrisonApiService,
+  csipApiService: CsipApiService,
+): RequestHandler {
+  const router = Router()
+
+  router.get('/service-not-enabled', (_req, res) => {
+    res.status(200)
+    res.render('service-not-enabled')
+  })
+
+  router.use(async (req, res, next) => {
+    const splitUrl = req.url.split('/').filter(Boolean)
+    if (
+      !uuidValidate(splitUrl[0] || '') &&
+      !req.url.endsWith('/start') &&
+      !req.url.includes('prisoner-image') &&
+      !req.url.includes('service-not-enabled')
+    ) {
+      try {
+        if (res.locals.feComponentsMeta?.caseLoads) {
+          res.locals.user.caseloads = res.locals.feComponentsMeta.caseLoads
+        }
+        if (res.locals.feComponentsMeta?.activeCaseLoad) {
+          res.locals.user.activeCaseLoadId = res.locals.feComponentsMeta.activeCaseLoad.caseLoadId
+        }
+        const refetchCaseloads = !res.locals.user.caseloads || res.locals.user.caseloads.length > 1
```

Comment:
This will refetch it even if it's come from the fe components if it has > 1 case load. Only needs the first check I think

Reply:
Done